sift/src/main/java/siftscience/android/AppStateCollector.java

```diff
@@ -169,7 +169,7 @@ private AndroidAppStateJson get() {
         List<String> addresses = new ArrayList<>();
         try {
             for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces();
-                 en.hasMoreElements(); ) {
+                 en != null && en.hasMoreElements(); ) {
                 NetworkInterface intf = en.nextElement();
                 for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses();
                      enumIpAddr.hasMoreElements(); ) {
```

Comment:
The [docs](https://docs.oracle.com/javase/7/docs/api/java/net/NetworkInterface.html#getInetAddresses()) for `getInetAddresses()` don't mention returning `null`.

Reply:
You are right -- based on the comment in the Android SDK 28 source, it seems to be something Google added recently. Their attempt to keep us all employed by fixing bugs they cause.